proposed/2022/PMUIPackageSourceMapping.md

```diff
@@ -0,0 +1,72 @@
+
+Two more scenarios could arise when a user tries to update a package if that package has an exisiting mapping. First, the mapping could be to a different source than the one chose from the package source dropdown when `Update` is clicked. In this case, the user would get a popup error message telling them that it cannot update because the package already has a mapping to a different source and ask if the user would like to update using the other source instead. The user could click `OK` to update using the other source or `Cancel`. The second scenario would be if the user tries to update a package that is already mapped to a different source (like the first scenario), but the source it is mapped to does not support the version they are trying to update to. In this case, the same popup error message would tell the user that the package is mapped to a different source, and that source does not support the version they are trying to install. The user would not be able to click `OK` in this case because restore would not be able to make this update.
+
+### Technical Explanation
+
+The mapping will not be written to the config until the user selects `OK` on the preview dialog popup. A mapping will only be written to the config if the user selects the checkbox in the details pane. <!--if not checked then key=* ? Does the user have to make a mapping? Maybe if user does not check the box and package source mapping is already enabled then they will get a message saying they should make a mapping--> 
+
+**Example 1** 
+
+```xml
+<PackageReference Include="Serilog" Version="11.0.0"/>
+```
+
+```xml
+<packageSourceMapping>
+    <packagesource key="nuget.org">
+        <package pattern="Serilog" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+**Result:**
+
+In this example, the user had `nuget.org` selected from the source dropdown. They clicked on the `Serilog` package on the installed tab, chose version `11.0.0`, and selected the checkbox to make a mapping from `Serilog` to `nuget.org`.
+
+## Drawbacks
+
+## Rationale and Alternatives
+
+On the previous spec, there was a pin icon next to the sources dropdown to pin a source instead of a checkbox in the details pane. I thought the pin was confusing since there was nothing to explain what it did. Also the same pin icon is used to different actions already in VS like to pin a file at the top of the screen. It would be confusing to have the same button do different tasks in different parts of VS.
```

Comment:
The older spec was based on an older name of "package namespaces & source pinning". We can use whatever affordance makes sense to "map a source".

Here are many other icons that are fair game to use: https://d3assets.azurewebsites.net/icons/Vswin2022

Anything surrounding the metaphor for a "relationship", "association", or even "map" may be worth pursuing. A checkbox in the details pane may have some cons to discoverability and UX of mapping batch packages on install/update.